client/src/connection.rs

```diff
@@ -318,21 +324,23 @@ impl ConnectionSender {
 
 pub struct Connection {
     addr: String,
+    auth_data: Option<Authentication>,
```

Comment:
It looks like this isn't used anywhere yet; were you planning on adding e.g. reconnect attempts or something where this is necessary? 

My instinct would be to avoid adding it to `Connection` until it's used to avoid the "field is never used" warning.

Reply:
I added it first when I was experimenting, but it's not needed, yes